gremlin-client/src/io/serializer_v3.rs

```diff
@@ -112,15 +112,42 @@ where
     if !val.is_empty() {
         let mut x = 0;
         while x < val.len() {
-            let key = get_value!(&val[x], Value::String)?;
+            //let key = get_value!(&val[x], Value::String)?;
+
+            let key = match &val[x] {
+                Value::String(s) => Ok(s.clone()),
+                Value::Object(_) => {
+                    let token_value: String = match reader(&val[x])? {
+                        GValue::Token(tk) => {
+                            let value = tk.value();
+                            Ok(value.clone())
+                        },
+                        _ => Err(GremlinError::Json(String::from(stringify!(o))))
+                    }?;
+
+                    Ok(token_value)
+                },
+                _ => Err(GremlinError::Json(String::from(stringify!($v)))),
```

Comment:
I believe this fragment can be improved. Suggestions are welcome. This has been my very first steps in Rust.